src/main/drivers/barometer_bmp085.c

```diff
@@ -371,4 +374,27 @@ static void bmp085_get_cal_param(void)
     bmp085.cal_param.md = (data[20] << 8) | data[21];
 }
 
+#if defined(BARO_EOC_GPIO)
+bool bmp085TestEOCConnected(const bmp085Config_t *config)
+{
+    uint32_t baroDeadline = 0;
+    uint32_t currentTime = micros();
+
+    baroDeadline += (UT_DELAY * 2); // wait twice as long as normal, just to be sure
+    bmp085_start_ut();
+    baroDeadline += micros();
+
+    while ((int32_t)(currentTime - baroDeadline) < 0) {
```

Comment:
using `delay(UT_DELAY / 1000 + 1 )` should be OK here. UT_DELAY has margin included.